lib/src/status.rs

```diff
@@ -305,19 +307,153 @@ pub(crate) async fn status(opts: super::cli::StatusOpts) -> Result<()> {
     let mut out = out.lock();
     let legacy_opt = if opts.json {
         OutputFormat::Json
+    } else if std::io::stdout().is_terminal() {
+        OutputFormat::HumanReadable
     } else {
         OutputFormat::Yaml
     };
     let format = opts.format.unwrap_or(legacy_opt);
     match format {
         OutputFormat::Json => serde_json::to_writer(&mut out, &host).map_err(anyhow::Error::new),
         OutputFormat::Yaml => serde_yaml::to_writer(&mut out, &host).map_err(anyhow::Error::new),
+        OutputFormat::HumanReadable => human_readable_output(&mut out, &host),
     }
     .context("Writing to stdout")?;
 
     Ok(())
 }
 
+fn human_readable_output(mut out: impl Write, host: &Host) -> Result<()> {
+    for (status_string, status) in [
+        ("staged", &host.status.staged),
+        ("booted", &host.status.booted),
+        ("rollback", &host.status.rollback),
+    ] {
+        if let Some(host_status) = status {
+            if let Some(image) = &host_status.image {
+                writeln!(
+                    out,
+                    "Current {} image: {}",
+                    status_string, image.image.image
+                )?;
+
+                let version = image
+                    .version
+                    .as_deref()
+                    .unwrap_or("No image version defined");
+                let timestamp = image
+                    .timestamp
+                    .as_ref()
+                    .map(|t| t.to_string())
+                    .unwrap_or_else(|| "No timestamp present".to_owned());
+                let transport = &image.image.transport;
+                let digest = &image.image_digest;
+
+                writeln!(out, "    Image version: {version} ({timestamp})")?;
+                writeln!(out, "    Image transport: {transport}")?;
+                writeln!(out, "    Image digest: {digest}")?;
+            } else {
+                writeln!(out, "Current {status_string} state is native ostree")?;
+            }
+        } else {
+            writeln!(out, "No {status_string} image present")?;
+        }
+    }
+    Ok(())
+}
+
+fn human_status_from_spec_fixture(spec_fixture: &str) -> Result<String> {
```

Comment:
Don't we get an unused function warning?

It may help to put everything here under a

```
#[cfg(test)]
mod tests {
...
}
```

as is done elsewhere in the code